### PR TITLE
Feature/3 ci cd

### PR DESCRIPTION
**Pull request by yevhen-komarov**
Изучение, настройка и тестирование GitHub Actions и GitHub Pages.
GitHub Pages: https://yevhen-komarov.github.io/dg-basic-react/

### CODE REVIEW

File: .github/workflows/develop-tests.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        node-version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+      - run: npm ci
```
Comment (mmospanenko) [inline .github/workflows/develop-tests.yml:25]: why npm when we're using yarn?

File: .github/workflows/develop-tests.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
```
Comment (mmospanenko) [inline .github/workflows/develop-tests.yml:23]: matrix? We don't need multiple versions to test with

File: .github/workflows/develop-tests.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x]
```
Comment (mmospanenko) [inline .github/workflows/develop-tests.yml:16]: use more specific version definitions

File: package.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
   "scripts": {
     "dev": "vite",
     "build": "tsc && vite build",
+    "build-gh-pages": "tsc && vite build --base=/dg-basic-react/",
```
Comment (mmospanenko) [inline package.json:8]: why not build? it is ok to have only one build destination for now

**Review by mmospanenko — CHANGES_REQUESTED** [review]
I think you simply copied and paste this from somewhere, you should get new knowledge, not just working result. And you have warnings - need to fix them all

File: .github/workflows/develop-tests.yml (hunk 1 above)
Comment (yevhen-komarov) [reply .github/workflows/develop-tests.yml]: fixed

File: .github/workflows/develop-tests.yml (hunk 2 above)
Comment (yevhen-komarov) [reply .github/workflows/develop-tests.yml]: fixed

File: .github/workflows/develop-tests.yml (hunk 3 above)
Comment (yevhen-komarov) [reply .github/workflows/develop-tests.yml]: fixed

**Review by mmospanenko — APPROVED** [review]
approved, added some comments/questions

File: package.json (hunk 4 above)
Comment (yevhen-komarov) [reply package.json]: not needed, removed tsc from scripts

**Pull request merged**